### PR TITLE
Add the feature to ignore metadata for OpenAPI metadata endpoints

### PR DESCRIPTION
**Pull request by adambarreiro**
This PR complements #557 by adding the support of `IgnoredMetadata` to OpenAPI metadata resources like `DefinedEntity`.

The implementation tries to avoid duplicating the already existing mechanism used by XML ignored metadata, by creating an intermediate struct that normalises both XML and OpenAPI flavors into a common type that can be understood by the same filtering function.

For that reason we have
- `normaliseXmlMetadata`: Transforms XML type into a normalised struct
- `normaliseOpenApiMetadata`: Transforms OpenAPI into a normalised struct

With the two normalised structs, we can use `filterSingleGenericMetadataEntry` that now it doesn't care about OpenAPI nor XML, it just contains the logic to filter.

Then we have two `filterSingleXmlMetadataEntry` and `filterSingleOpenApiMetadataEntry` that calls the normalising function and the filtering.

### CODE REVIEW

File: govcd/metadata_v2.go [hunk 1]
```diff
@@ -954,43 +978,55 @@ func filterMetadata(allMetadata *types.Metadata, href, objectName string, metada
...
+	}
+	isFiltered := filterSingleGenericMetadataEntry(normalisedEntry, metadataToIgnore)
+	if isFiltered {
+		return nil, fmt.Errorf("the metadata entry with key '%s' and value '%v' is being ignored", key, metadataEntry.TypedValue.Value)
+	}
+	return metadataEntry, nil
+}
+
+// filterSingleGenericMetadataEntry filters a single metadata entry given a slice of metadata that needs to be ignored. It doesn't
 // alter the input metadata, but returns a copy of the filtered metadata.
```
Comment (Didainius) [inline govcd/metadata_v2.go:1007]: This comment does not look exactly valid anymore as it returns `bool`, not a copy of filtered metadata.

**Review by Didainius — APPROVED** [review]
I think I am generally good. I've got one comment. RDE test passed for me.

**Review by lvirbalas — APPROVED** [review]
Great job!

File: govcd/metadata_openapi.go [hunk 2]
```diff
@@ -250,3 +269,30 @@ func addOpenApiMetadata(client *Client, endpoint, objectId string, metadataEntry
...
+// Ignore OpenAPI Metadata feature
+// ---------------------------------------------------------------------------------------------------------------------
+
+// normaliseOpenApiMetadata transforms OpenAPI metadata into a normalised structure
+func normaliseOpenApiMetadata(objectType, name string, metadataEntry *types.OpenApiMetadataEntry) (*normalisedMetadata, error) {
+	return &normalisedMetadata{
+		ObjectType: objectType,
+		ObjectName: name,
+		Key:        metadataEntry.KeyValue.Key,
+		Value:      fmt.Sprintf("%v", metadataEntry.KeyValue.Value.Value),
```
[the remark this answers is not in the document]
Comment (adambarreiro) [reply govcd/metadata_openapi.go:283]: That's a very good suggestion, I have added the following unit tests:
- `Test_normaliseXmlMetadata`
- `Test_normaliseOpenApiMetadata`
in two new files: `metadata_v2_unit_test.go` and `metadata_openapi_unit_test.go` respectively.

Both unit tests check all possible metadata type conversions.

**Review by adezxc — APPROVED** [review]

**Pull request merged**